src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -888,6 +888,16 @@ private void EnsureSessionHandleExists(WinHttpRequestState state)
 
                 HttpResponseMessage responseMessage = WinHttpResponseParser.CreateResponseMessage(state, _doManualDecompressionCheck);
                 state.Tcs.TrySetResult(responseMessage);
+
+                if((responseMessage.StatusCode == HttpStatusCode.MultipleChoices ||
+                    responseMessage.StatusCode == HttpStatusCode.MovedPermanently ||
+                    responseMessage.StatusCode == HttpStatusCode.Redirect ||
+                    responseMessage.StatusCode == HttpStatusCode.RedirectMethod ||
+                    responseMessage.StatusCode == HttpStatusCode.RedirectKeepVerb) &&
+                   state.RequestMessage.RequestUri.Scheme == Uri.UriSchemeHttps &&
+                   responseMessage.Headers.Location?.Scheme == Uri.UriSchemeHttp) {
+                    WinHttpTraceHelper.Trace("WinHttpHandler.SendAsync: Insecure https to http redirect blocked.");
```

Comment:
It would be nice if we had EventSourceTrace style tracing available in WinHttpHandler (similar to CurlHandler, etc.), but that is probably beyond the scope of this PR.